src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -671,11 +736,31 @@ private sealed class CoherentState
         {
             private readonly ConcurrentDictionary<string, CacheEntry> _stringEntries = new ConcurrentDictionary<string, CacheEntry>(StringKeyComparer.Instance);
             private readonly ConcurrentDictionary<object, CacheEntry> _nonStringEntries = new ConcurrentDictionary<object, CacheEntry>();
+
+#if NET9_0_OR_GREATER
+            private readonly bool _useStringAltLookup;
+            private readonly ConcurrentDictionary<string, CacheEntry>.AlternateLookup<ReadOnlySpan<char>> _stringAltLookup;
+
+            public CoherentState()
+            {
+                _useStringAltLookup = _stringEntries.TryGetAlternateLookup<ReadOnlySpan<char>>(out _stringAltLookup);
```

Comment:
Can this just use GetAlternateLookup instead?

Reply:
see 4f601494c54bc080224f8b2bb540a004ae0da751